code/game/machinery/sci_bombardment.dm

```diff
@@ -0,0 +1,257 @@
+	playsound(src, 'sound/effects/podwoosh.ogg', 80, 0.25)
+	scibomb.forceMove(dest)
+	playsound(scibomb, 'sound/effects/bamf.ogg', 95, 0.25, 75, 1, 0, 0, FALSE, TRUE) //Minimum impact sound in the odd event Toxins doesn't send a bomb
+	scibomb.toggle_valve()
+	dest = initial(dest)
+	targetdest = initial(dest)
+	tcoords = initial(tcoords)
+	scibomb = initial(scibomb)
+	update_icon()
+	. = TRUE
+
+/**
+* Visual proc, temporarily disables interacting
+* 
+* Triggered after the ui_act's "target" and "abort" to
+* add a delay between selecting GPS coordinates and 
+* cancelling a TTV launch.
+*/
+/obj/machinery/sci_bombardment/proc/reset_lam() 
+	target_delay = !target_delay
+	update_icon()
+	if(target_delay)
+		spawn(100)
+			reset_lam()
+	else
+		playsound(src, 'sound/items/scanner_match.ogg', 100, 1)
+		return
+
+/obj/machinery/sci_bombardment/ui_interact(mob/user, ui_key = "lam", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		var/lam_window_height = clamp(325 + GLOB.GPS_list.len * 24, 325, 700)
+		ui = new(user, src, ui_key, "Lam", name, 470, lam_window_height, master_ui, state)
+		ui.open()
+
+/obj/machinery/sci_bombardment/ui_data(mob/user)
+	var/list/data = list()
+	data["tcoords"] = tcoords
+	data["countdown"] = countdown
+	data["locked"] = locked
+	data["loaded"] = scibomb
+	data["stopcount"] = stopcount
+	if(locked)
+		return data
+	if(tcoords)
+		data["tcoords"] = tcoords
+		data["targetdest"] = targetdest
+
+	var/list/signals = list()
+	data["signals"] = list()
+
+	for(var/gps in GLOB.GPS_list)
+		var/obj/item/gps/G = gps
+		var/turf/pos = get_turf_global(G) // yogs - get_turf_global instead of get_turf
+		if(G.emped || !G.tracking || pos.z != 5)
+			continue
+		var/list/signal = list()
+		signal["entrytag"] = G.gpstag //GPS name
+		signal["coords"] = "[pos.x], [pos.y], [pos.z]"
+		signals += list(signal)
+	data["signals"] = signals
+	return data
+
+/obj/machinery/sci_bombardment/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("lock")//Check for RD/Silicon access. Lock/Unlock console if valid
+			if(iscyborg(usr) || isAI(usr))
+				locked = !locked
+				radio.talk_into(src, "Controls [locked ? "locked" : "unlocked"] by [usr].",)
+			else 
+				var/mob/M = usr
+				var/obj/item/card/id/I = M.get_idcard(TRUE)
+				if(check_access(I) && (30 in I.access))
+					locked = !locked
+					radio.talk_into(src, "Controls [locked ? "locked" : "unlocked"] by [I.registered_name].",)
+				else
+					to_chat(usr, "<span class='warning'>Access denied. Please seek assistance from station AI or Research Director.</span>")
+			update_icon()
+			. = TRUE
+		if("count")//Prompts user to change countdown timer (Minimum based on var/mincount)
+			if(locked)
+				return
+			var/a = text2num(stripped_input(usr, "Set a new countdown timer. (Minimum [mincount])", name, mincount))
```

Comment:
slap a max arg on this. Some enterprising shitters will eventually find the list of href exploits and force this to hit negative and instantly send the bomb somehow.